reference-implementation/lib/readable-stream.js

```diff
@@ -86,273 +89,126 @@ class ReadableStream {
   }
 
   pipeTo(dest, { preventClose, preventAbort, preventCancel } = {}) {
-    // brandcheck
+    if (IsReadableStream(this) === false) {
+      return Promise.reject(streamBrandCheckException('pipeTo'));
+    }
+    if (IsWritableStream(dest) === false) {
+      return Promise.reject(
+        new TypeError('ReadableStream.prototype.pipeTo\'s first argument must be a WritableStream'));
+    }
 
     preventClose = Boolean(preventClose);
     preventAbort = Boolean(preventAbort);
     preventCancel = Boolean(preventCancel);
 
-    const source = this;
+    const reader = AcquireReadableStreamDefaultReader(this);
+    const writer = AcquireWritableStreamDefaultWriter(dest);
```

Comment:
Why do the brand checks above return a rejected promise (instead of throwing synchronously), when `AcquireReadableStreamDefaultReader` and `AcquireWritableStreamDefaultWriter` might throw synchronously anyway if either stream is already locked? Shouldn't either all of these throw, or all of these be promises?